src/bgen/CodeBlocks/IfBlock.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+using System.IO;
+
+public class IfBlock : CodeBlock {
+	List<CodeBlock> ElseIfBlocks = new ();
+	CodeBlock? ElseBlock = null;
+	public IfBlock (string condition)
+	{
+		HeaderText = "if (" + condition + ")";
+	}
+
+	public IfBlock (string condition, List<ICodeBlock> blocks)
+	{
+		HeaderText = "if (" + condition + ")";
+		Blocks.AddRange (blocks);
+	}
+
+	public void AddElseIf (string condition, List<ICodeBlock> blocks)
```

Comment:
If you make this look like this:
```
public IfBlock AddElseIf (string condition, List<ICodeBlock> blocks)
{
    ElseIfBlocks.Add (new CodeBlock ($"else if ({condition})", blocks));
    return this;
}
```
You get two things - string interpolation reads a little better and the fluent interface lets you write:
```
var test = new IfBlock ("someTest", blocks)
                          .AddElse ("someOtherTest", otherBlocks)
                          .AddElse ("someOtherOtherTest", otherOtherBlocks)
                          .AddElse ("other code);
```

Reply:
If you make the blocks a params:

```cs
public IfBlock AddElseIf (string condition, params ICodeBlock[] blocks)
{
    ElseIfBlocks.Add (new CodeBlock ($"else if ({condition})", blocks));
    return this;
}
```

you can do:

```cs
var test = new IfBlock ("someTest", blocks)
                          .AddElseIf ("someOtherTest", 
							new Block ("Console.WriteLine(var1);"),
							new Block ("Console.WriteLine(var2);"))
                          .AddElseIf ("someOtherOtherTest",
							new Block ("Console.WriteLine(var3);"))
                          .AddElse ("other code);
```